Readme
What is the program about?
Record is the name of my large-scale programming project. The program’s main goal is to develop a PC platform that allows users to produce files using a program.  Record would allow users to not only create an unlimited number and type of files, but also to edit, view, and delete them through the program, similar to how Samsung Notes and Notes apps on the iPhone operate. 

How does the program work?

The program is very easy to use, with clear instructions that help the users operate the program. There are four main parts to the program. The main parts are: creating, viewing, editing and deleting external files. 

Creating files:
In the main menu, there is an option to create external files. After you enter the right letter associated with this option, you are asked the name of the files you want to create, and what you want to write. You are only allowed to write one line at a time. After the process, the external with the entered title name and its contents are saved in the “Files” menu on the Replit program. 

Viewing files:
After you have created the file, you now have different options to what to do with the program. In the option “view a file”, you can open external files that are specifically created on this Replit program. Other external files that are created or used outside this program will not be accessible. You need to create an external file first in order to view the file.

Editing files:
After you have created the external file and possibly view it, you can also edit the document. In the option “edit a file”, you can open and edit the external file. First, you will be asked the name of the external file you want to edit, you will then view the file and its contents, and you will be asked again if you want to add anything to the file or not. If you have answered “yes”, you will now be able to add one line to the file. To enter several lines, you need to repeat the process for every line you want to add because you are only allowed to add one line at a time. A solution to this is to access the external file in the “Files” menu on the left side of the Replit program, where you can add as many lines as you want, and it will be automatically saved in the program. 

Deleting files:
If you want to delete an external file in the program, you can also access the option in the menu. 
In the option “delete a file”, you can delete a file. You will be asked the name of the file you want to delete. Then, the file and its content will be opened in view mode, and the program will again ask if you want to delete the file in order to confirm the process. After you have entered yes, the file and its contents will be automatically deleted in the program. 

Bonus options
Downloading files:
In this option of the Replit program, you can also download the external file if you want to access it on your desktop or on your phone. In the “Files” menu on the left side of the Replit program page, click the three-dots on the right side of the file you want to download. There you will be given a bunch of options with one of the options including “Download”. Click on the option and the process will begin automatically. Now, you can also access the files on your desktop. 

Closing the program:
In the menu, you are also given an option to “Close the program”. In this option, you can close the program regardless of where you are in the program, and all the files and its content will remain saved in the program.

Help:
In this option, you are given important rules and guidelines on how the program works and how to use the program correctly. 

Target Audience
My target audience of this program are those people who not only want to create notes, similar to how they are created in Samsung Notes or in the Notes app in iPhone, but this program is also for those who want to create different types of external files, ranging from text (.txt) files to PY (.py) files, and to also access them to view, edit, delete or download them in their desktops. The lack of an easy-to-use platform on PC that gives users various alternatives such as producing external files and storing them on their desktop, which can aid users in their daily lives, is an issue. For example, a person wants to create a text(.txt) file to download and save on his PC for a specific activity, but he doesn't know what platform to use. His solution is Record, a platform that allows users to create a text file (.txt) for their needs, as well as edit, download, and save the external file to their PC.


changelog
# changelog
All notable changes of this project will be documented here 

## [0.0.1] 2021-11-23
### initial commit
- function for creating files has been created
- an intro is made for the program
- a menu is made, lists all the valid actions

## [0.0.2] 2021-11-24
### Added
- function for viewing files has been created

## [0.0.3] 2021-11-25
### Added
- a loop is added for the program

## [0.0.4] 2021-11-26
### Added
- function for deleting files has been created
- the intro for the program is edited
- function is created for editing files

## [0.0.4] 2021-11-30
### Added
- code is corrected through the Python Style Guide

## [0.0.5] 2021-12-09
### Changed
- Python Style Guide corrections

## [0.0.5] 2021-12-13
### Changed
- Intro for the program is updated

## [0.0.5] 2021-12-14
### Added
- Module named "help" is created consisting of guidelines for the program

## [0.0.6] 2021-12-15
### Added
- A Readme file is created and added to the program, which consists of basic information of the program.

## [0.0.7] 2022-01-04
### Added
- Python Style Guide corrections

help.py
def long_line():
    """ This creates a line between the output to seperate code """
    print("------------------------------------------------------------------")


def help():
    print("""
                                    HELP

        Some basic rules of the program:

        Enter the letter associated with the option in the menu to
        access that part of the program.
        For example: to access "a: create a file" in the menu,
        you need to enter the letter "a" when asked
        ,"What would you like to do today",
        and it will then take you to that part of the program.

        When asked a question in the program, enter "yes" or "no"
        for the program to process the request

        Create a file first in order to access other features
        such as editing or viewing a File

        You can only write one line when creating a File.
        To add more, go to "edit a file" to add one line at a time.

        You can not have two files with the same name.
        New file created with the same name will overwrite the existing
        file's contents.

        When entering a file name, the name should be matching the
        letter case of the file.""")
    long_line()


edit_files.py
import main


def note_3():
    """ This function allows user to
        edit an existing file """
    try:
        filename = input("Which file do you want to edit?: ")
        print("\n")
        filename_2 = open(filename, "r")
        print(filename + ":")
        print(filename_2.read())
        print("\n")
        add_note = input("Do you want to add more? ")
        if add_note == "yes":
            add = input("What do you want to add:\n")
            with open(filename, "a+") as file_object:
                file_object.write("\n")
                data = file_object.read(100)
                if len(data) > 0:
                    file_object.write("\n")
                file_object.write(add)
            print("\n")
            filename_3 = open(filename, "r")
            print(filename + ":")
            print(filename_3.read())
            print("\n")
            main.line()
            main.intro()
        if add_note == "no":
            main.line()
            main.intro()
    except FileNotFoundError:
        print("File does not exists. returning to main menu")
        main.line()
        main.intro()


main.py
# Course: CS 30
# Period: 2
# Date created: 23/11/21
# Date last modified: 01/12/21
# Name: Humzah Zahid Malik
# Description: Large Scale program


import os
import help


def long_line():
    """ This creates a line between the output to seperate code """
    print("------------------------------------------------------------------")


# Creates a menu of valid actions in this program


valid_actions = ["a: create a file", "b: view a file", "c: edit a file",
                 """d: delete a file""", """e: Close the program""",
                 """f: Help"""]


def main_menu():
    """ Prints the introduction of the program """
    print("""Hello there, and welcome to Record!

Record is a Python-based program for creating, viewing, editing, and deleting
external files.

You can create a variety of files on this platform, ranging from
text(.txt) files to PY(.py) files, and
you can create an unlimited number of files.

Important:
Enter "f" for "Help", which will guide to the basic guidelines of the program.

   """)
    print("\n")


def intro():
    """ Prints all of the valid actions in a menu """
    print(""" Choose an action: """)
    for action in valid_actions:
        print(f" {action} ")
    input_1 = input("What would you like to do today? ")
    if input_1 == "a":
        notes_1()
    if input_1 == "b":
        note_2()
    if input_1 == "c":
        note_3()
    if input_1 == "d":
        note_4()
    if input_1 == "e":
        print(" You have closed the program ")
        quit()
    if input_1 == "f":
        help.help()
    else:
        print("invalid action")
        long_line()


def notes_1():
    """ This function creates new files """
    filename = input("What is the title of the note: ")
    file_name = filename
    character = input("What do you want to note? ")
    print("\n")
    with open(file_name, 'w') as file_object:
        file_object.write(character)
    filename_2 = open(filename, "r")
    print(file_name + ":")
    print(filename_2.read())
    print("\n")
    long_line()
    intro()


def note_2():
    """ This functions opens files in view mode """
    try:
        filename = input("Which file do you want to open?: ")
        print("\n")
        filename_2 = open(filename, "r")
        print(filename + ":")
        print(filename_2.read())
        print("\n")
        long_line()
        intro()
    except FileNotFoundError:
        print("File does not exists. returning to main menu")
        long_line()
        intro()


def note_3():
    """ This function allows user to
        edit an existing file """
    try:
        filename = input("Which file do you want to edit?: ")
        print("\n")
        filename_2 = open(filename, "r")
        print(filename + ":")
        print(filename_2.read())
        print("\n")
        add_note = input("Do you want to add more? ")
        if add_note == "yes":
            add = input("What do you want to add:\n")
            with open(filename, "a+") as file_object:
                file_object.write("\n")
                data = file_object.read(100)
                if len(data) > 0:
                    file_object.write("\n")
                file_object.write(add)
            print("\n")
            filename_3 = open(filename, "r")
            print(filename + ":")
            print(filename_3.read())
            print("\n")
            long_line()
            intro()
        if add_note == "no":
            long_line()
            intro()
    except FileNotFoundError:
        print("File does not exists. returning to main menu")
        long_line()
        intro()


def note_4():
    """ This functions allows user to delete an existing file """
    try:
        filename = input("Which file do you want to delete?: ")
        print("\n")
        filename_2 = open(filename, "r")
        print(filename + ":")
        print(filename_2.read())
        print("\n")
        delete = input("Do you want to delete file: ")
        if delete == "yes":
            os.remove(filename)
            print("file : '" + filename + "'" + " is deleted")
            long_line()
            intro()
        if delete == "no":
            long_line()
            intro()
    except FileNotFoundError:
        print("File does not exists. returning to main menu")
        long_line()
        intro()


while True:
    main_menu()
    intro()
